### PR TITLE
feat: support preview setting tags

### PR DESCRIPTION
**Pull request by rzhao271**
Adds two tags called "preview" and "experimental" to the suggestions list, and recommends config writers to use a new onExP tag instead to suggest that a setting can be part of an A/B experiment.

### CODE REVIEW

**Comment by rzhao271** [comment]
experimentalFeature and previewFeature tag suggestions in the Settings editor, though they are not next to each other because the suggestions are sorted alphabetically

File: src/vs/workbench/services/configuration/browser/configurationService.ts [hunk 1]
```diff
@@ -1360,7 +1360,11 @@ class UpdateExperimentalSettingsDefaults extends Disposable implements IWorkbenc
 		const allProperties = this.configurationRegistry.getConfigurationProperties();
 		for (const property of properties) {
 			const schema = allProperties[property];
-			if (!schema?.tags?.includes('experimental')) {
+			const tags = schema?.tags;
+			// Many experimental settings refer to in-development or unstable settings.
+			// onexperiment more clearly indicates that the setting could be
+			// part of an experiment.
+			if (!tags || (!tags.includes('experimental') && !tags.includes('onexperiment'))) {
```
Comment (sbatten) [inline src/vs/workbench/services/configuration/browser/configurationService.ts:1367]: I'm not sure I fully understand this migration strategy, but I also have a slight preference for `onExp` or something that really drives that this is part of our Exp platform.

Comment (rzhao271) [reply src/vs/workbench/services/configuration/browser/configurationService.ts:1367]: 1. Allow both tag:experimental and tag:onExp settings to be ExP-assignable for now.
2. File issues for current tag:experimental users. I believe a lot of third-party extensions already use it in the non-ExP sense, meaning that the current mixup is mostly an internal or first-party issue.
3. Disallow tag:experimental settings to be ExP-assignable.

**Comment by sandy081** [comment]
Are we deprecating `experimental` tag in favour of `onExP` ? If so why?

**Comment by rzhao271** [comment]
@sandy081 this PR allows the preview and experimental tags to refer to non-stable settings, and creates a new onExP tag to refer to settings that can be part of ExP assignments/experiments. Otherwise, we would leave the experimental tag as-is, but then have to create verbose tags like previewQuality and experimentalQuality to talk about non-stable settings.

**Review by sbatten — APPROVED** [review]

**Review by songjianet — APPROVED** [review]

**Pull request merged**